### ISSUE TITLE
FR: Allow URIs in headless mode

### ISSUE THREAD

**Issue by dangayle** [craftcms/cms#5265]
### Description

Currently, enabling headless mode [disables UR Format settings for sections and category groups](https://docs.craftcms.com/v3/config/config-settings.html#headlessmode). However, many of us use CraftCMS as a headless CMS and still need URIs but would like the other benefits of using headless mode. 

- Gatsby users typically pass in a pathname (e.g., the uri) when using the gatsby-node.js `createPage` method. Without this, we would need an entirely extraneous URI constructor.
- Live previews in Gatsby also require a URI
- Headless mode won't work with plugins such as SEOmatic, since it needs to match the URI which gets turned off when headlessMode is enabled.

So, basically a lot of the people who are using CraftCMS as a headless CMS are unable to use headless mode because we still need URIs.

**Comment by khalwat** [comment]
This is coming in Craft CMS 3.4 -> https://github.com/craftcms/cms/issues/4520#issuecomment-531040417
